### ISSUE TITLE
Round error when casting float to decimal

### ISSUE THREAD

**Issue by andygrove** [apache/arrow-datafusion#4071]
**Describe the bug**

## Spark

```
scala> spark.sql("select cast(0.06+0.01 as decimal(18,2))").show
+------------------------------------+
|CAST((0.06 + 0.01) AS DECIMAL(18,2))|
+------------------------------------+
|                                0.07|
+------------------------------------+
```

## DataFusion

```
❯ select cast(0.06+0.01 as decimal(18,2));
+-------------------------------+
| Float64(0.06) + Float64(0.01) |
+-------------------------------+
| 0.06                          |
+-------------------------------+
```

**To Reproduce**
See above

**Expected behavior**

**Additional context**
Discovered as part of https://github.com/apache/arrow-datafusion/issues/4024

**Comment by andygrove** [comment]
filed related issue in arrow-rs: https://github.com/apache/arrow-rs/issues/2997

**Comment by tustvold** [comment]
Fixed by #4199 

```
❯ select cast(0.06+0.01 as decimal(18,2));
+-------------------------------+
| Float64(0.06) + Float64(0.01) |
+-------------------------------+
| 0.07                          |
+-------------------------------+
1 row in set. Query took 0.008 seconds.
```
